crates/core/tedge_api/src/entity_store.rs

```diff
@@ -751,6 +775,71 @@ impl EntityTree {
             vec![]
         }
     }
+
+    fn get_parent(&self, topic_id: &EntityTopicId) -> Result<&EntityTopicId, Error> {
+        let parent = self
+            .try_get(topic_id)?
+            .parent
+            .as_ref()
+            .unwrap_or(&self.main_device);
+        Ok(parent)
+    }
+
+    pub fn update_parent(
+        &mut self,
+        topic_id: &EntityTopicId,
+        new_parent: &EntityTopicId,
+    ) -> Result<&EntityMetadata, Error> {
+        if topic_id == new_parent {
+            return Err(Error::InvalidSelfParent(new_parent.clone()));
+        }
```

Comment:
Isn't this a sub-case of the cycle check done a few line bellow with ?

```
   if self.ancestors(new_parent)?.contains(&topic_id) {
     ...
   }
```

Reply:
Since `ancestors()` of an entity starts with the parent and doesn't include that entity itself, this special case was required.